### ISSUE TITLE
A `match` statement within an `array` produces `Squiz.Arrays.ArrayDeclaration.NoKeySpecified`

### ISSUE THREAD

**Issue by ivan-wolf** [squizlabs/PHP_CodeSniffer#3502]
**Description**
The `default` option of the `match` statement will result in the following error:
`No key specified for array entry; first entry specifies key`

**Code sample**
```php
[
    'foo' => match ($anything) {
        'foo' => 'bar',
        default => null,
    },
];
```

**Expected behavior**
Should not result in an error.

**Versions:**
 - OS: Ubuntu 20.04
 - PHP: `8.0`
 - PHPCS: `3.6.2`
 - Standard: `Squiz`

**Comment by gsherwood** [comment]
Thanks for the bug report. I've added match support to the sniff and the fix will be in 3.7.0
